packages/dashbuilder/kie-soup-dataset/kie-soup-dataset-api/src/main/java/org/dashbuilder/dataset/json/ExternalDefJSONMarshaller.java

```diff
@@ -46,9 +48,10 @@ public void fromJson(ExternalDataSetDef def, JsonObject json) {
         var headers = json.getObject(HEADERS);
         var accumulate = json.getBoolean(ACCUMULATE);
         var type = json.getString(TYPE);
+        var join = json.getArray(JOIN);
 
-        if (isBlank(url) && isBlank(content)) {
-            throw new IllegalArgumentException("External Data Sets must have \"url\" or \"content\" field");
+        if (isBlank(url) && isBlank(content) && (join == null || join.length() == 0)) {
```

Comment:
```suggestion
        if (isBlank(url) && isBlank(content) && isBlank(join)) {
```
Can we use `isBlank` for `join` too?

Reply:
It will not work because `isBlank` is only for Strings and join is a JSON array